app/views/users/_form.html.erb

```diff
@@ -20,8 +20,8 @@
 
   <div class='tab-content'>
     <div class='tab-pane active' id='primary'>
-
-      <%= text_f f, :login, :disabled => f.object.login == 'admin' %>
+      <% can_edit_login =   User.current.can?(:edit_users, @user) || ( f.object.id == User.current.id  && User.current.internal? ) %>
```

Comment:
Please move this logic to the User model (as a validation) & test it, that'd help ensuring this doesn't break in the future (methods in views are prone to break without anyone noticing)